wikimedia/wikipedia-ios
# [YiR] Fix Explore profile button

**Phabricator:** N/A

### Notes
This PR fixes a race condition that I think only appeared once https://github.com/wikimedia/wikipedia-ios/pull/5044 and https://github.com/wikimedia/wikipedia-ios/pull/5056 were merged.

I noticed that ExploreViewController's lazy yirCoordinator was stuck with nil, because it's `yirDataController` would throw a `coreDataStoreUnavailable` error upon instantiation. Likely when a YearInReviewDataController is instantiated early on in the app lifecycle, the core data store isn't ready yet. Adjusting these a bit to make the `yirDataController` computed on Explore and Article (so that it tries again each time), and relaxing the laziness a bit for `yirCoordinator`, seems to help.

### Test Steps
1. Fresh install app in correct device region with. correct device language. Confirm Explore profile button tap works.
2. Enable YiR feature flag. Background, foreground. Leave Explore tab and come back. Confirm feature announcement and badge appear. Confirm tapping into intro slide clears out badge as it did before. Confirm survey still presents after the last slide.
3. Repeat steps 1-2 for Article view.
